consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/ExecutionPlanRepositoryImpl.scala

```diff
@@ -135,36 +136,30 @@ class ExecutionPlanRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends
     }
   }
 
-  override def getDataSources(value: String, access: Option[String])(implicit ec: ExecutionContext): Future[Array[String]] ={
+  override def getDataSources(value: String, access: Option[AccessValue])(implicit ec: ExecutionContext): Future[Array[String]] = {
     val readResult = db.queryStream[String](
       """
-        FOR d in dataSource
-        |FOR dp in depends
-        |FILTER dp._from == @planId AND dp._to == d._id
-        |RETURN d.uri
-        |""".stripMargin
-        .stripMargin,
-      Map("planId" -> "executionPlan".concat("/").concat(value)
-      )
+        |FOR ds IN 1..1
+        |OUTBOUND DOCUMENT('executionPlan', @planId) depends
+        |RETURN ds.uri
+        |""".stripMargin,
+      Map("planId" -> s"executionPlan/$value")
     )
 
     val writeResult = db.queryStream[String](
       """
-        FOR d in dataSource
-        |FOR aff in affects
-        |FILTER aff._from == @planId AND aff._to == d._id
-        |RETURN d.uri
-        |""".stripMargin
-        .stripMargin,
-      Map("planId" -> "executionPlan".concat("/").concat(value)
-      )
+        |FOR ds IN 1..1
+        |OUTBOUND DOCUMENT('executionPlan', @planId) affects
+        |RETURN ds.uri
+        |""".stripMargin,
+      Map("planId" -> s"executionPlan/$value")
     )
 
-    val totalResult = readResult.zip(writeResult).map{case(listOne,listTwo) => listOne ++ listTwo}
+    val totalResult = readResult.zip(writeResult).map{ case(listOne,listTwo) => listOne ++ listTwo }
 
-    if(access.isEmpty)
+    if (access.isEmpty)
       totalResult.map(_.toArray)
-    else if(access.equals("read"))
+    else if (access.get.equals(AccessValue.read))
       readResult.map(_.toArray)
     else
       writeResult.map(_.toArray)
```

Comment:
Sorry I didn't make it clear. One of the biggest benefits of using enums is that you can apply exhaustive pattern-matching instead of `if-else` blocks. Also it's a known best practice to use sealed traits with case objects instead of native Scala enums, primarily to be able to use _exhaustive_ pattern-matching. See for instance [this blog post](https://medium.com/@yuriigorbylov/scala-enumerations-hell-5bdba2c1216). Or take a look at `OnDBExistsAction` class in our project.

Reply:
Another observation - you execute both _depends_ and _affects_ queries on every method call regardless of what was actually requested. It has better be done lazily, so only necessary queries are executed.

Reply:
@wajda what should one return when no access type is provided? I was providing both affects and depends by default and hence calculating it again
do you mean, we should use lazy val? 
Thanks

Reply:
sharing a code snippet for search
`def getDataSourceResultFromActionType(
    actionType: DataSourceActionType,
    readResult: Future[Stream[String]],
    writeResult : Future[Stream[String]]): Future[Stream[String]] = actionType match {
    case Read => readResult
    case Write => writeResult
  }`

Reply:
Following your suggestions, this is what i came up with .
If accessType is empty, we return overview results.
What do you think of the the below implementation 
thanks
`
if(access.isEmpty)
      db.queryStream[String](
        """
          |FOR ds IN 1..1 OUTBOUND DOCUMENT ('executionPlan', '20e4a7d1-610b-459a-97b9-cff7aac4b1ab')  GRAPH 'overview'
          | RETURN ds.uri
          |""".stripMargin,
        Map("planId" -> s"$execPlanId")
      ).map(_.toArray)
    else
    access.get match {
      case Read =>  db.queryStream[String](
        """
          |FOR ds IN 1..1
          |OUTBOUND DOCUMENT('executionPlan', @planId) depends
          |RETURN ds.uri
          |""".stripMargin,
        Map("planId" -> s"$execPlanId")
      ).map(_.toArray)

      case Write => db.queryStream[String](
        """
          |FOR ds IN 1..1
          |OUTBOUND DOCUMENT('executionPlan', @planId) affects
          |RETURN ds.uri
          |""".stripMargin,
        Map("planId" -> s"$execPlanId")
      ).map(_.toArray)
    }
  }`

Reply:
`lazy val` could be a solution, yes. Your last snippet would work as well. Yet another solution would be a dynamically generated query. Whatever you prefer.

Regarding handling empty `accessType` I think that's OK. Although personally I'd rather pass `Seq[AccessType]` instead of `Option[AccessType]` and would return empty result if no access type is provided. But that's might be a matter of taste, so feel free to decide.

The code style-wise there are two things that caught my attention in your last code piece:
1. `s"$execPlanId"` is just `execPlanId`
2. `access.get` / `access.isEmpty` - using the combination of those method is considered a bad practice. Try using more idiomatic approach to _Option_ handling. E.g.
```scala
access
  .map({
    case Read => ...
    case Write => ...
  })
  .getOrElse({
    ...
  })
```

Reply:
And one more little thing - in your AQL using `depends, affects` is more readable and understandable than `GRAPH 'overview'`. Technically it's the same thing, but in this context where you have three similar queries that only differ in selected edges, it just feels more natural to stick with edge names instead of graph names.

Reply:
I added my last snippet
You are absolutely right about the graph overview , didnt strike me that time. Thanks for the tip.

I kept it as option and i am returning both results if no access type is provided instead of an empty list. Hope that works ?
Changed the variable names to proper names.